src/components/lab/Menu/MenuItem.tsx

```diff
@@ -0,0 +1,215 @@
+import * as React from 'react';
+
+import {useFloatingTree, useListItem} from '@floating-ui/react';
+import {ChevronLeft, ChevronRight} from '@gravity-ui/icons';
+
+import {useForkRef} from '../../../hooks';
+import {BUTTON_ICON_SIZE_MAP} from '../../Button/constants';
+import {Icon} from '../../Icon';
+import {useDirection} from '../../theme';
+import {block} from '../../utils/cn';
+
+import {MenuContext} from './MenuContext';
+import {MenuItemContext} from './MenuItemContext';
+import type {
+    MenuItemButtonProps,
+    MenuItemComponentElementType,
+    MenuItemComponentProps,
+    MenuItemLinkProps,
+    MenuItemProps,
+    MenuProps,
+} from './types';
+import {isComponentType} from './utils';
+
+import './MenuItem.scss';
+
+function isMenuItemComponentProps<T extends MenuItemComponentElementType>(
+    p: MenuItemProps<T>,
+): p is MenuItemComponentProps<Exclude<T, undefined>> {
+    return p.component !== undefined;
+}
+
+const b = block('menu2-item');
+
+export const MenuItem = React.forwardRef(
```

Comment:
Why not [ListItemView](https://github.com/gravity-ui/uikit/blob/main/src/components/lab/ListItemView/ListItemView.tsx) from lab?